src/main/java/com/stripe/model/BalanceTransaction.java

```diff
@@ -146,7 +148,16 @@ public String getType() {
 	public void setType(String type) {
 		this.type = type;
 	}
+	
+	public void setSourceObject(HasId sourceObject) {
+		this.sourceObject = sourceObject;
+	}
 
+	@SuppressWarnings("unchecked")
+	public <O> O getSourceObject() {
+		return (O) sourceObject;
+	}
```

Comment:
Ha. So that's actually pretty much what my very first implementation was based on, remember? See https://github.com/stripe/stripe-java/pull/361 for more details.

The problem with it was that it's not compatible with webhooks b/c data structure that is submitted to webhook handler doesn't support expandable fields and it was failing b/c of that.

That's the reason I closed that pull request and created this one following the new approach. This one works with both API endpoints (i.e. when you request to expand balance transaction source object) and webhook handlers (i.e. when data structure that does not follow expandable source object format is submitted to webhook handler and you need to deserialize Event that contains nested balance transaction).